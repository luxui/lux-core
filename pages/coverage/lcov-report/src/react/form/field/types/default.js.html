<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/react/form/field/types/default.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">src/react/form/field/types</a> default.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'; // `React` must be in scope when using JSX
&nbsp;
import registry from '../../../../lib/componentRegistry';
&nbsp;
import '../wrapper';
&nbsp;
function defaultFieldComponent(props) {
  const Wrapper = registry('Lux.Form.FieldWrapper');
  const { name, title, value = 'no value' } = props;
  const style = {
    background: 'darkRed',
    color: 'white',
    padding: '1ex 3ex',
  };
&nbsp;
  return (
    &lt;Wrapper className={props.className} style={style}&gt;
      &lt;pre&gt;No field type component for: {title}&lt;/pre&gt;
      &lt;pre&gt;{JSON.stringify({ name, title, value }, null, 4)}&lt;/pre&gt;
    &lt;/Wrapper&gt;
  );
}
defaultFieldComponent.propTypes = {
  className: React.PropTypes.string,
  name: React.PropTypes.string,
  title: React.PropTypes.string,
  value: React.PropTypes.string,
};
&nbsp;
registry('Lux.Form.Field.Default', defaultFieldComponent, false);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Tue Feb 28 2017 08:27:25 GMT-0500 (EST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
