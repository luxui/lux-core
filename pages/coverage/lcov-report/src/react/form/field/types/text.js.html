<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/react/form/field/types/text.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">src/react/form/field/types</a> text.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'; // `React` must be in scope when using JSX
&nbsp;
import registry from '../../../../lib/componentRegistry';
&nbsp;
import '../wrapper';
&nbsp;
import { addType } from './';
&nbsp;
function textFieldComponent(props, context) {
  const Wrapper = registry('Form.FieldWrapper');
  const { name, title, value } = props;
&nbsp;
  const inputAttrs = {
    defaultValue: value,
    id: name,
    name,
    onChange: (event) =&gt; {
      context.setState({
        ...context.state,
        [name]: event.target.value,
      });
    },
  };
&nbsp;
  return (
    &lt;Wrapper className={props.className}&gt;
      &lt;label htmlFor={name}&gt;{title}&lt;/label&gt;
      &lt;input {...inputAttrs} /&gt;
    &lt;/Wrapper&gt;
  );
}
textFieldComponent.contextTypes = {
  setState: React.PropTypes.func,
  state: React.PropTypes.object,
};
textFieldComponent.propTypes = {
  className: React.PropTypes.string,
  name: React.PropTypes.string,
  title: React.PropTypes.string,
  value: React.PropTypes.string,
};
&nbsp;
const name = 'Form.Field.Text';
&nbsp;
addType('text', name);
registry(name, textFieldComponent, false);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Feb 24 2017 11:21:10 GMT-0500 (EST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
