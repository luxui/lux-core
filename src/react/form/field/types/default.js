import React from 'react'; // `React` must be in scope when using JSX

import registry from '../../../../lib/componentRegistry';

import '../wrapper';

function defaultFieldComponent(props) {
  const Wrapper = registry('Lux.Form.FieldWrapper');
  const { name, title, value = 'no value' } = props;
  const style = {
    background: 'darkRed',
    color: 'white',
    padding: '1ex 3ex',
  };

  return (
    <Wrapper className={props.className} style={style}>
      <pre>No field type component for: {title}</pre>
      <pre>{JSON.stringify({ name, title, value }, null, 4)}</pre>
    </Wrapper>
  );
}
defaultFieldComponent.propTypes = {
  className: React.PropTypes.string,
  name: React.PropTypes.string,
  title: React.PropTypes.string,
  value: React.PropTypes.string,
};

registry('Lux.Form.Field.Default', defaultFieldComponent, false);
