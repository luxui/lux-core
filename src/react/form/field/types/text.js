import React from 'react'; // `React` must be in scope when using JSX

import registry from '../../../../lib/componentRegistry';

import '../wrapper';

import { addType } from './';

function textFieldComponent(props, context) {
  const Wrapper = registry('Form.FieldWrapper');
  const { name, title, value } = props;

  const inputAttrs = {
    defaultValue: value,
    id: name,
    name,
    onChange: (event) => {
      context.setState({
        ...context.state,
        [name]: event.target.value,
      });
    },
  };

  return (
    <Wrapper className={props.className}>
      <label htmlFor={name}>{title}</label>
      <input {...inputAttrs} />
    </Wrapper>
  );
}
textFieldComponent.contextTypes = {
  setState: React.PropTypes.func,
  state: React.PropTypes.object,
};
textFieldComponent.propTypes = {
  className: React.PropTypes.string,
  name: React.PropTypes.string,
  title: React.PropTypes.string,
  value: React.PropTypes.string,
};

const name = 'Form.Field.Text';

addType('text', name);
registry(name, textFieldComponent, false);
