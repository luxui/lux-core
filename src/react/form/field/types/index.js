import { isString } from '../../../../lib/is';

import './default';

const types = {};

function addType(type, id) {
  if (!isString(type)) {
    // eslint-disable-next-line max-len
    throw new Error(`The "type" must be a string; ${typeof type} provided (${type}).`);
  }

  if (!isString(id)) {
    // eslint-disable-next-line max-len
    throw new Error(`The "id" must be a string; ${typeof id} provided (${id}).`);
  }

  types[type] = id;
}

function getType(type) {

  return types[type] || 'Form.Field.Default';
}

export {
  addType,
  getType,
};
