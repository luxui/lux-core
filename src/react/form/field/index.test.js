import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';

import registry from '../../../lib/componentRegistry';

import './';
import { addType } from './types';

const shallow = ReactTestUtils.createRenderer();

const FormField = registry('Form.Field');

const defaultField = registry('Form.Field.Default');
const fieldTypes = [
  // ['checkbox', 'Form.Field.Checkbox'],
  // ['color', 'Form.Field.Color'],
  // ['date', 'Form.Field.Date'],
  // ['datetime', 'Form.Field.Datetime'],
  // ['datetime-local', 'Form.Field.DatetimeLocal'],
  // ['email', 'Form.Field.Email'],
  // ['file', 'Form.Field.File'],
  // ['hidden', 'Form.Field.Hidden'],
  // ['month', 'Form.Field.Month'],
  // ['number', 'Form.Field.Number'],
  // ['password', 'Form.Field.Password'],
  // ['radio', 'Form.Field.Radio'],
  // ['range', 'Form.Field.Range'],
  // ['select', 'Form.Field.Select'],
  // ['search', 'Form.Field.Search'],
  // ['tel', 'Form.Field.Tel'],
  ['text', 'Form.Field.Text'],
  // ['textarea', 'Form.Field.Textarea'],
  // ['time', 'Form.Field.Time'],
  // ['url', 'Form.Field.Url'],
  // ['week', 'Form.Field.Week'],
];

function componentName(obj) {

  return obj && obj.type && obj.type.name;
}

describe('Form.Field', function () {
  it('should exist; and should be a function', function () {
    expect(typeof FormField).toBe('function');
  });

  it('should render the "default" component', function () {
    const component = shallow.render(<FormField type={''} />);

    expect(componentName(component)).toBe(defaultField.name);
  });

  fieldTypes.forEach(([type, name]) => {
    const Component = registry(name);

    if (!Component) return;

    it(`should render ${type} form field`, function () {
      const component = shallow.render(<FormField type={type} />);

      expect(componentName(component)).toBe(Component.name);
    });
  });

  it('should render custom form field types when registered', function () {
    const name = 'customFieldType';

    registry(name, () => (
      <div>Custom Form Field</div>
    ))
    addType(name, name);

    const result = renderer.create(
      <FormField type={name} />
    );
    const tree = result.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
