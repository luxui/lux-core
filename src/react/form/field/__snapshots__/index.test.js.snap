// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form.Field should render custom form field types when registered 1`] = `
<div>
  Custom Form Field
</div>
`;
